specs/admin_wave1_spec.rb

```diff
@@ -0,0 +1,230 @@
+
+  end
+
+  it "raises error for invalid dates" do
+    hotel = Hotel::Admin.new
+    start_date = Date.new(2018,05,9)
+    end_date = Date.new(2018,05,06)
+    #res = hotel.make_reservation(start_date,end_date,2)
+    proc{hotel.total_cost(start_date,end_date)}.must_raise ArgumentError
+  end
+
+end
+
+describe "list_of_available_rooms" do
+
+  it "raises error if given date is invalid" do
+    hotel = Hotel::Admin.new
+    proc{hotel.get_reservations(2)}.must_raise ArgumentError
+
+  end
+
+  it "returns the list of empty rooms available on a date" do
+    hotel = Hotel::Admin.new
+
+    hotel.make_reservation(Date.new(2018,05,5),Date.new(2018,05,7))
+    hotel.make_reservation(Date.new(2018,05,6),Date.new(2018,05,8))
+    hotel.make_reservation(Date.new(2018,05,5),Date.new(2018,05,9))
+
+    hotel.list_of_available_rooms(Date.new(2018,05,4)).length.must_equal 20
+    hotel.list_of_available_rooms(Date.new(2018,05,5)).length.must_equal 19
+    hotel.list_of_available_rooms(Date.new(2018,05,6)).length.must_equal 19
+    hotel.list_of_available_rooms(Date.new(2018,05,8)).length.must_equal 20
+
+  end
+
+end
+
+describe "reserve_list_available_rooms" do
+
+  it "returns the list of all reservations" do
+    hotel = Hotel::Admin.new
+    start_date = Date.new(2018,05,03)
+    end_date = Date.new(2018,05,06)
+    reservation = hotel.reserve_available_rooms(start_date,end_date)
+    reservation.class.must_equal Array
+
+  end
+
+  it "returns the newly created reservation" do
+    hotel = Hotel::Admin.new
+    start_date = Date.new(2018,05,03)
+    end_date = Date.new(2018,05,06)
+    before_length = hotel.all_reservations.length
+    reservation = hotel.reserve_available_rooms(start_date,end_date)
+    reservation.length.must_equal before_length + 1
+
+  end
+
+  it "raises an error there are no available rooms" do
+    hotel = Hotel::Admin.new
+    hotel.list_of_available_rooms(Date.new(2018,5,3))
+    20.times do
+      hotel.reserve_available_rooms(Date.new(2018,5,3),Date.new(2018,5,6))
+    end
+
+    proc{hotel.reserve_available_rooms(Date.new(2018,5,3),Date.new(2018,5,6))}.must_raise Exception
+  end
+
+
+  it "raises an error for invalid date range" do
+    hotel = Hotel::Admin.new
+    start_date = Date.new(2018,5,10)
+    end_date =Date.new(2018,5,6)
+    proc{hotel.reserve_available_rooms(start_date,end_date)}.must_raise ArgumentError
+
+  end
+
+end
+
+describe "create_block" do
+
+  it "creates an instance of BlockReservation class" do
+    hotel = Hotel::Admin.new
+    block = hotel.create_block(Date.new(2018,5,5), Date.new(2018,5,9),"one",150)
+    block.must_be_instance_of Array
+  end
+
+  it " creates a block of rooms " do
+    hotel = Hotel::Admin.new
+    block = hotel.create_block(Date.new(2018,5,5), Date.new(2018,5,9),"one",150)
+
+    block[0].block_id.must_equal "one"
+    block[0].status.must_equal true
+    block[0].room_number.must_be_instance_of Hotel::Room
+    block[0].discount_rate.must_equal 150
+
+  end
+
```

Comment:
You need more through testing here, including if there aren't enough rooms available in the date range due to reservations.